lib/shoryuken/extensions/active_job_concurrent_send_adapter.rb

```diff
@@ -0,0 +1,42 @@
+# ActiveJob docs: http://edgeguides.rubyonrails.org/active_job_basics.html
+# Example adapters ref: https://github.com/rails/rails/tree/master/activejob/lib/active_job/queue_adapters
+module ActiveJob
+  module QueueAdapters
+    # == Shoryuken concurrent adapter for Active Job
+    #
+    # This adapter sends messages asynchronously (ie non-blocking) and allows
+    # the caller to set up handlers for both success and failure
+    #
+    # To use this adapter, set up as:
+    #
+    # adapter = ActiveJob::QueueAdapters::ShoryukenConcurrentSendAdapter.new
+    # adapter.success_handler = ->(job, options) { StatsD.increment(job.class.name + "success") }
+    # adapter.error_handler = ->(err, (job, options)) { StatsD.increment(job.class.name + "failure") }
+    #
+    # config.active_job.queue_adapter = adapter
+    class ShoryukenConcurrentSendAdapter < ShoryukenAdapter
+      attr_accessor :error_handler
```

Comment:
I'm open to moving this as constructor requirement but the advantage to leaving this as an `attr_accessor` is callers can switch the handlers for different tasks.  For instance, we might care about an error handler for a mission critical code path and not in others.  Allowing callers to change this at run time has value for us but I'm open to alternatives.

Reply:
> Allowing callers to change this at run time has value for us but I'm open to alternatives.

@allcentury Would change at runtime be thread-safe?

Reply:
@phstc - probably not, I'm also naively thinking of my own use case, where a k8s cron task is run.  I can change this in the task though so I think using dependency injection (like your added commit does) is the way to go.